src/impy/kinematics.py

```diff
@@ -540,12 +519,37 @@ def boost_lab_to_cms(self, event):
 
 
 class CenterOfMass(EventKinematics):
-    def __init__(self, energy, particle1, particle2):
+    def __init__(self, ecm, particle1, particle2):
         impy_config["user_frame"] = "center-of-mass"
-        super()._init(ecm=energy, particle1=particle1, particle2=particle2)
+        super().__init__(ecm=ecm, particle1=particle1, particle2=particle2)
+
+
+class TotalEnergy(float):
+    def __new__(cls, value):
+        return float.__new__(cls, value)
+
+
+class KinEnergy(float):
+    def __new__(cls, value):
+        return float.__new__(cls, value)
+
+
+class Momentum(float):
+    def __new__(cls, value):
+        return float.__new__(cls, value)
```

Comment:
Sorry, I realized that we don't even need to define `__new__`,

```
class TotalEnergy(float): pass
```
is sufficient here.

Reply:
Just remark: Without further methods added, these Tag types revert to float on any arithmetic operation.
```
en = TotalEnergy(10)
type(en)  # __main__.TotalEnergy
type(en + 1.0)  # float
type(en * 1.0)  # float
en += 1.0
type(en) # float
```
To avoid surprises like this, we would have to implement the correspinding operators to return instances of our class instead of float.

Reply:
Another option is to not derive from float after all, to avoid calling methods that don't do what we expect they do. The more I think about it, the less I also like that the repr does not indicate that this is a tagged float.

Let me add a patch to impy.util with a proper tagged float type.

Reply:
@jncots I added a patch with a new TaggedFloat that avoids the issues.

Reply:
I also added some tests for the new behavior of FixedTarget, which were missing.